### PR TITLE
Add ValidatingAdmissionPolicy warning example

### PR DESCRIPTION
**Pull request by cici37**
Fix https://github.com/kubernetes/website/issues/40952

### CODE REVIEW

**Comment by cici37** [comment]
/assign @sftim 
Thank you!

**Comment by T-Lakshmi** [comment]
Many of the checks failed as netlify build failed due to timeout. Could you try to trigger the build again with using

git commit --amend --no-edit && git push --force-with-lease

File: content/en/blog/_posts/2020-09-03-warnings/index.md [hunk 1]
```diff
@@ -320,6 +320,17 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
...
 and required to pass with a zero exit code in order for the CI job to succeed.
 
+## Validating Admission Policy
+
+We introduced [Validating Admission Policy](/docs/reference/access-authn-authz/validating-admission-policy) in 1.26 as an alpha feature.
+Starting in Kubernetes v1.27, a field `validationAction` is added in ValidatingAdmissionPolicyBinding which is used to declare
+how Validations of the referenced ValidatingAdmissionPolicy are enforced. When it sets to `Warn`, a validation failure is reported to the request client
+in HTTP Warning headers, with a warning code of 299.
+
+<img alt="kubectl applying a manifest file with ValidatingAdmissionPolicy and validationAction is set to `Warn`, displaying a warning message and exiting with a non-zero exit code."
```
Comment (sftim) [inline content/en/blog/_posts/2020-09-03-warnings/index.md:330]:
```suggestion
<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
```

File: content/en/blog/_posts/2020-09-03-warnings/index.md [hunk 2]
```diff
@@ -320,6 +320,17 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
 This could be used in a CI job to apply manifests to a current server,
 and required to pass with a zero exit code in order for the CI job to succeed.
 
+## Validating Admission Policy
+
```
Comment (sftim) [inline content/en/blog/_posts/2020-09-03-warnings/index.md:324]:
```suggestion

_Added in 2023_

```

**Review by sftim — COMMENTED** [review]
How about these ideas?

**Comment by sftim** [comment]
> Many of the checks failed as netlify build failed due to timeout. Could you try to trigger the build again with using
> 
> git commit --amend --no-edit && git push --force-with-lease

Even better: rebase against _main_ (see https://github.com/kubernetes/website/issues/35780#issuecomment-1548286534)

**Comment by cici37** [comment]
All comments are addressed. PTAL when you have time @sftim , Thank you :)

File: content/en/blog/_posts/2020-09-03-warnings/index.md [hunk 3]
```diff
@@ -320,6 +320,19 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
...
+_Added in 2023_
```
Comment (tengqm) [inline content/en/blog/_posts/2020-09-03-warnings/index.md:325]: Shall we add these as more formal documents?

Comment (sftim) [reply content/en/blog/_posts/2020-09-03-warnings/index.md:325]: We do have https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/ already.

File: content/en/blog/_posts/2020-09-03-warnings/index.md [hunk 4]
```diff
@@ -320,6 +320,19 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
...
+
+We introduced [Validating Admission Policy](/docs/reference/access-authn-authz/validating-admission-policy) in 1.26 as an alpha feature.
+Starting in Kubernetes v1.27, a field `validationAction` is added in ValidatingAdmissionPolicyBinding which is used to declare
+how Validations of the referenced ValidatingAdmissionPolicy are enforced. When it sets to `Warn`, a validation failure is reported to the request client
+in HTTP Warning headers, with a warning code of 299.
+
+<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
+src="kubectl-warnings-vap.png"
+style="width:637px;max-width:100%;">
```
Comment (sftim) [inline content/en/blog/_posts/2020-09-03-warnings/index.md:334]: This isn't rendering - see https://deploy-preview-41332--kubernetes-io-main-staging.netlify.app/blog/2020/09/03/warnings/

Please fix that and check that the [local preview](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#preview-locally) looks OK.

File: content/en/blog/_posts/2020-09-03-warnings/index.md (hunk 3 above)
Comment (cici37) [reply content/en/blog/_posts/2020-09-03-warnings/index.md:325]: As Tim mentioned, we have a separate place to document the feature and here is the place we maintain warnings I guess :)

File: content/en/blog/_posts/2020-09-03-warnings/index.md (hunk 4 above)
Comment (cici37) [reply content/en/blog/_posts/2020-09-03-warnings/index.md]: H @sftim , I was inspecting in the preview page and looks like the png file wasn't been built together hence it cannot find the file. Is there any recommendation on this? Thank you!

Comment (cici37) [reply content/en/blog/_posts/2020-09-03-warnings/index.md]: @sftim friendly ping to see if it is expected behavior(not build png file for preview) or if I did something wrong :)

File: content/en/blog/_posts/2020-09-03-warnings/index.md [hunk 5]
```diff
@@ -320,6 +320,19 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
...
+_Added in 2023_
+
+We introduced [Validating Admission Policy](/docs/reference/access-authn-authz/validating-admission-policy) in 1.26 as an alpha feature.
+Starting in Kubernetes v1.27, a field `validationAction` is added in ValidatingAdmissionPolicyBinding which is used to declare
+how Validations of the referenced ValidatingAdmissionPolicy are enforced. When it sets to `Warn`, a validation failure is reported to the request client
+in HTTP Warning headers, with a warning code of 299.
+
+<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
+     src="kubectl-warnings-vap.png"
+     style="width:637px;max-width:100%;">
```
Comment (sftim) [inline content/en/blog/_posts/2020-09-03-warnings/index.md:334]:
```suggestion
<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
     src="kubectl-warning-vap.png"
     style="width:637px;max-width:100%;">
```

Try this. Or, rename the image.

**Review by sftim — COMMENTED** [review]
Thanks. Please fix the typo and also - because this PR predates the v1.28 release - rebase against _main_.

File: content/en/blog/_posts/2020-09-03-warnings/index.md (hunk 4 above)
Comment (sftim) [reply content/en/blog/_posts/2020-09-03-warnings/index.md]: BTW sorry about the wait - I have a fairly long backlog of PRs etc where people have mentioned me, and I tend to use an oldest-first approach to find these

**Review by nate-double-u — COMMENTED** [review]
Hi @cici37, there are a couple updates suggested for this edit. Are these changes still relevant? Would you be able to make the update and rebase if so?

File: content/en/blog/_posts/2020-09-03-warnings/index.md (hunk 5 above)
Comment (cici37) [reply content/en/blog/_posts/2020-09-03-warnings/index.md:334]: Thank you so much for catching this! I have updated. :)

**Comment by cici37** [comment]
> Hi @cici37, there are a couple updates suggested for this edit. Are these changes still relevant? Would you be able to make the update and rebase if so?

Thanks for the ping. I have updated and addressed all comments and squashed the commits. It should be ready to merge. :)

**Review by nate-double-u — COMMENTED** [review]
Thanks for the updates @cici37!

@liggitt I'd like to ping you too, given you wrote the original article.

/lgtm

**Comment by sftim** [comment]
@liggitt comments: 

> if you wanted to help people make a connection to the new thing, a single line + link at the end after a footer clearly separate from the blog pointing people at the actual docs for a new thing seems like it would be fine as well, rather than new paragraphs and screenshots

I think the screenshots sell it to me. It's too big a change if we have screenshots. @cici37 would you be willing to write a new blog article about updates to warnings since the original thing, and then revise the original article to direct readers there to read on?

You could also add a single line linking to docs (not a blog article) about the documented example. Your choice.

**Comment by divya-mohan0209** [comment]
@cici37 : Would you be willing to make a separate blog PR per the suggestion by @sftim ?

**Comment by divya-mohan0209** [comment]
Hi @cici37 , thanks for the PR. However, since it has been ~2 months without a response, I'll be closing this one off. We encourage you to submit a separate PR for the blog instead of reopening the PR, in case you'd like to continue working on this particular subject. 

/close

**Pull request closed without merging**